test/helpers/kubectl.go

```diff
@@ -223,13 +229,55 @@ func (kub *Kubectl) WaitforPods(namespace string, filter string, timeout time.Du
 	return true, nil
 }
 
+// WaitForServiceEndpoints waits up until timeout seconds have elapsed for all
+// endpoints in the specified namespace that match the provided JSONPath filter
+// to have their port equal to the provided port. Returns true if all pods achieve
+// the aforementioned desired state within timeout seconds. Returns false and
+// an error if the command failed or the timeout was exceeded.
+func (kub *Kubectl) WaitForServiceEndpoints(namespace string, filter string, service string, port string, timeout time.Duration) (bool, error) {
+	body := func() bool {
+		var jsonPath = fmt.Sprintf("{.items[?(@.metadata.name =='%s')].subsets[0].ports[0].port}", service)
+		data, err := kub.GetEndpoints(namespace, filter).Filter(jsonPath)
+
+		if err != nil {
+			kub.logger.Errorf("could not get service endpoints: %s", err)
+			return false
+		}
+		log.Infof("data: %s", data.String())
```

Comment:
This log message is not clear, and should be `Debugf`

Reply:
This was leftover from debugging, will remove in next upload.

Reply:
Fixed.